app/upgrade_cudos.go

```diff
@@ -2029,7 +2029,8 @@ func DoGenesisAccountMovements(genesisData *GenesisData, cudosCfg *CudosMergeCon
 						return fmt.Errorf("failed to move delegated amount %s of %s from %s to %s: %w", delegatedAmount, validatorAddr, accountMovement.SourceAddress, accountMovement.DestinationAddress, err)
 					}
 				}
-				remainingAmountToMove = remainingAmountToMove.Sub(delegatedAmount)
+				newBalance := remainingAmountToMove.Sub(delegatedAmount)
+				remainingAmountToMove = &newBalance
```

Comment:
Even though this is correcty, it is literally callinbg for a problem later on.
Well, actually to avoid confusion, the actual issue is at [line 2012 here](https://github.com/fetchai/fetchd/pull/388/files#diff-3b0d03fd0d353dbb228a94b4720a117e15e86e275e76fb26cec0236cedf82d2eR2012), I just cound not put comment on that line because of lame github web UI limitation

That thing is, that the `remainingAmountToMove` variable is a pointer pointing exactly at the same `sdk.Int` instance as `accountMovement.Amount` data member.
Now, this kicks open door for potentially someone later on modifying the code, not realising where `remainingAmountToMove` points to, and simply doing following thing (note that little `*` de-referencing operator there):
```go
*remainingAmountToMove = something
```
That is practically impossible to catch by human eye during review, and it will cause havoc, which would be really hard to pinpoint what is going on.

I would suggest to create new copy-by-value at line 2012:
```go
		remainingAmountToMove := sdk.NewIntFromBigInt(accountMovement.Amount.BigInt())
		if sourceDelegations, exists := genesisData.delegations.Get(accountMovement.SourceAddress); exists {
			for i := range sourceDelegations.Iterate() {
				validatorAddr, delegatedAmount := i.Key, i.Value

				if delegatedAmount.GTE(remainingAmountToMove) {
					// Split delegation
					err := moveGenesisDelegation(genesisData, accountMovement.SourceAddress, accountMovement.DestinationAddress, validatorAddr, *remainingAmountToMove, manifest, "")
					if err != nil {
						return fmt.Errorf("failed to move delegated amount %s of %s from %s to %s: %w", delegatedAmount, validatorAddr, accountMovement.SourceAddress, accountMovement.DestinationAddress, err)
					}

					break
				} else {
					// Move entire delegation
					err := moveGenesisDelegation(genesisData, accountMovement.SourceAddress, accountMovement.DestinationAddress, validatorAddr, delegatedAmount, manifest, "")
					if err != nil {
						return fmt.Errorf("failed to move delegated amount %s of %s from %s to %s: %w", delegatedAmount, validatorAddr, accountMovement.SourceAddress, accountMovement.DestinationAddress, err)
					}
				}
				remainingAmountToMove = remainingAmountToMove.Sub(delegatedAmount)
			}
		}
```